src/helpers/chartHelpers.tsx

```diff
@@ -3,38 +3,47 @@ import { isNumber } from "lodash";
 import { ChartThreshold, MultiSeriesLineChartOptions } from "./chartOptions";
 import { resolveColor } from "./colors";
 import { LayoutContext } from "../components";
+import { TimeSeriesDataPoint } from '../components/presentational/TimeSeriesChart/TimeSeriesChart';
 
-export function createLineChartDefs(layoutContext: LayoutContext, gradientKey: string, seriesLineColors: string[], multiSeriesLineChartOptions?: MultiSeriesLineChartOptions) {
-    let nDomainMin: number | undefined = undefined;
-    let nDomainMax: number | undefined = undefined;
+export function createLineChartDefs(
+    layoutContext: LayoutContext, 
+    gradientKey: string, 
+    seriesLineColors: string[], 
+    multiSeriesLineChartOptions: MultiSeriesLineChartOptions | undefined,
+    dataKeys: string[],
+    data: TimeSeriesDataPoint[] ) {
```

Comment:
The main change here is that the dataKeys and data need to be passed in so that each line's min/max can be calculated.  I broke this out into single lines because it was getting extremely long.